contributor-advice.md

```diff
@@ -0,0 +1,107 @@
+# Some advice for Alpakka contributors
+
+## API design
+
+### Flows
+
+When designing Flows, consider adding an extra field to the in- and out-messages which is passed through. A common
+use case we see, is committing a Kafka offset after passing data to another system.
+
+### Settings
+
+Most technologies will have a couple of configuration settings that will be needed for all Flows. Create case classes 
+collecting these settings instead of passing them in every method.
+
+Create a default instance in the companion object with good defaults which can be updated via `copy` or `withXxxx` methods.
+
+As Java can't use the `copy` methods, add `withXxxx` methods to specify certain fields in the settings instance.
+
+In case you see the need to support reading the settings from `Config`, offer a method taking the `Config` instance so
+that the user can apply a proper namespace.
+Refrain from using `akka.stream` as prefix, prefer `alpakka` as root namespace.
```

Comment:
This sentence is very true but has little to do with Settings. I'd move it to a separate section for this kind of general stuff

Reply:
Ah, it talks about the Config namespace. No the package names. I make it more specific.